observation/store.go

```diff
@@ -44,10 +47,14 @@ func (store *Store) SaveAll(ctx context.Context, observations []*models.Observat
 	instanceObservations := mapObservationsToInstances(observations)
 
 	for instanceID, observations := range instanceObservations {
-		dimensionIds, err := store.dimensionIDCache.GetNodeIDs(ctx, instanceID)
-		if err != nil {
-			store.reportError(ctx, instanceID, "failed to get dimension node id's for batch", err)
-			return results, err
+		var dimensionIds map[string]string
+		var err error
+		if store.getGraphDimensionID {
+			dimensionIds, err = store.dimensionIDCache.GetNodeIDs(ctx, instanceID)
+			if err != nil {
+				store.reportError(ctx, instanceID, "failed to get dimension node id's for batch", err)
+				return results, err
+			}
 		}
```

Comment:
If the flag is disabled, then we are not populating the dimensionIds map. I imagine we want to add an else here, and inside the else block we will manually create the dimensionIds (I think this was the whole point of this work?)

Reply:
Update: I just realised that the Trello card says that we should pass an empty map if the flag is false. So please, ignore my previous comment.